sdk/azidentity/managed_identity_client.go

```diff
@@ -34,14 +34,14 @@ const (
 	identityServerThumbprint = "IDENTITY_SERVER_THUMBPRINT"
 	headerMetadata           = "Metadata"
 	imdsEndpoint             = "http://169.254.169.254/metadata/identity/oauth2/token"
+	miResID                  = "mi_res_id"
```

Comment:
How come this is still needed (in what scenarios)?

I can't easily tell from the logic below where we'd use this one vs the MSI one. 
Both seem to be wrapped around `if id.idKind() == miResourceID`

Reply:
We still need this for App Service. Each platform we support has a different API, so the query parameter we need to set depends on the platform. `id` in the line you pasted is the value of the parameter i.e. the resource ID.